src/Raven.Client/Documents/Changes/DatabaseChanges.cs

```diff
@@ -466,12 +470,15 @@ private async Task DoWork()
                 }
                 catch (Exception e)
                 {
+
                     //We don't report this error since we can automatically recover from it and we can't
                     // recover from the OnError accessing the faulty WebSocket.
                     try
                     {
                         ConnectionStatusChanged?.Invoke(this, EventArgs.Empty);
 
+                        _serverNode = await _requestExecutor.HandleServerDown(_url.AbsoluteUri, _serverNode, _nodeIndex, e).ConfigureAwait(false);
```

Comment:
I think you should move this to be around `_client.ConnectAsync`, instead.
This can happen because the connection just broke, and we don't want to report server down for that.